### PR TITLE
Remove Epetra

### PR DESCRIPTION
**Pull request by mperego**
This PR removes all the code related to Epetra and all the tests associated to Epetra.
Before the removal, the Epetra tests that did not have a Tpetra analog have been converted to Tpetra tests.

The build_type has been removed, so "Build Type" needs to be removed from input yaml files.

### CODE REVIEW

File: tests/landIce/FO_GIS/CMakeLists.txt [hunk 1]
```diff
@@ -88,28 +87,27 @@ ENDIF()
...
-                   ${CMAKE_CURRENT_BINARY_DIR}/input_fo_gis_unstruct_restart.yaml)
-    add_test(${testName}_Restart ${Albany.exe} input_fo_gis_unstruct_restart.yaml)
-    set_tests_properties(${testName}_Restart
-                         PROPERTIES
-                         LABELS "LandIce;Epetra;Forward"
-                         RUN_SERIAL TRUE
-                         FIXTURES_REQUIRED UnstructRestart)
-  endif()
+# First run
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/input_fo_gis_unstruct.yaml
```
Comment (bartgol) [inline tests/landIce/FO_GIS/CMakeLists.txt:92]: I think the unstruct and unstructT tests are now very similar (if not the same). Can we prune one of them?

File: tests/landIce/FO_GIS/input_fo_humboldt_frosch_effect_press.yaml [hunk 2]
```diff
@@ -153,7 +152,7 @@ ANONYMOUS:
         Method: Ioss
         Number Of Time Derivatives: 0
         Restart Index: 1
-        Use Serial Mesh: false
+        Use Serial Mesh: ${USE_SERIAL_MESH}
```
Comment (bartgol) [inline tests/landIce/FO_GIS/input_fo_humboldt_frosch_effect_press.yaml:155]: Thanks for fixing this too!

**Review by bartgol — APPROVED** [review]
Nice! I think there's one FO_GIS test that is redundant, but other than that it's good.

Perhaps at a later stage, we could rename all the blahT.yaml file to blah.yaml (or viceversa). There's a bit of a mix right now, due to leftover from epetra-vs-tpetra duplication of some tests...

File: src/evaluators/scatter/PHAL_ScatterResidual_Def.hpp [hunk 3]
```diff
@@ -280,7 +280,7 @@ evaluateFields(typename Traits::EvalData workset)
 #endif
 
 #ifdef ALBANY_KOKKOS_UNDER_DEVELOPMENT
```
[the remark this answers is not in the document]
Comment (mcarlson801) [reply src/evaluators/scatter/PHAL_ScatterResidual_Def.hpp:282]: I don't think we do. I've removed them in a few places in a previous PR and these ones in gather and scatter will go away with my next uvm-free PR.

**Comment by mcarlson801** [comment]
It looks like some of the tests are failing with this branch on weaver:

```     
 46 - landIce_FO_GIS_Unstructured_Restart (Failed)
 61 - landIce_FO_GIS_SensSMBwrtBeta_Restart (Failed)
 64 - landIce_FO_GIS_Humboldt_LaplacianPriorSampling (Failed)
 99 - corePDEs_SideSetLaplacian_3D (Failed)
```

The FO_GIS tests are failing to find decomposed meshes:

```
Exodus Library Warning/Error: [ex__check_file_type]
    ERROR: Could not open file '/home/maxcarl/weaver/builds/albany-remove_epetra/tests/landIce/FO_GIS/gis_unstruct.exo.4.3', error = No such file or directory.
```

I think this is something Luca just fixed for other tests. Do we need to add something to the inputs for these tests?

The SideSetLaplacian_3D test is failing with the following error:

```
99: ************************************************************************
99: -- Nonlinear Solver Step 49 --
99: ||F|| = 2.500e-01  step = 1.000e+00  dx = 0.000e+00
99: ************************************************************************
99:
99: Ifpack2::ILUT::Compute: zero pivot encountered! Replacing with rownorm and continuing...(You may need to set the parameter 'fact: absolute threshold'.)
99: Ifpack2::ILUT::Compute: zero pivot encountered! Replacing with rownorm and continuing...(You may need to set the parameter 'fact: absolute threshold'.)
99: Ifpack2::ILUT::Compute: zero pivot encountered! Replacing with rownorm and continuing...(You may need to set the parameter 'fact: absolute threshold'.)
99: Ifpack2::ILUT::Compute: zero pivot encountered! Replacing with rownorm and continuing...(You may need to set the parameter 'fact: absolute threshold'.)
99:
99:   *******************************************************
99:   ***** Belos Iterative Solver:  Block Gmres
99:   ***** Maximum Iterations: 100
99:   ***** Block Size: 1
99:   ***** Residual Test:
99:   *****   Test 1 : Belos::StatusTestImpResNorm<>: (2-Norm Res Vec) / (2-Norm Res0), tol = 0.0001
99:   *******************************************************
99:   Iter   0, [ 1] :    1.000000e+00
99:   Belos::BlockGmresSolMgr::solve(): Warning! NaN has been detected!
99: WARNING: NOX::Direction::Newton::compute() - Linear solve failed to achieve convergence - using the step anyway since "Rescue Bad Newton Solve" is true
99: NOX::Solver::LineSearchBased::iterate - using recovery step for line search
```

Not sure why this would fail but not the 2D version.

**Comment by mperego** [comment]
> It looks like some of the tests are failing with this branch on weaver:
> 
> ```
>  46 - landIce_FO_GIS_Unstructured_Restart (Failed)
>  61 - landIce_FO_GIS_SensSMBwrtBeta_Restart (Failed)
>  64 - landIce_FO_GIS_Humboldt_LaplacianPriorSampling (Failed)
>  99 - corePDEs_SideSetLaplacian_3D (Failed)
> ```
> 
> The FO_GIS tests are failing to find decomposed meshes:
> 
> ```
> Exodus Library Warning/Error: [ex__check_file_type]
>     ERROR: Could not open file '/home/maxcarl/weaver/builds/albany-remove_epetra/tests/landIce/FO_GIS/gis_unstruct.exo.4.3', error = No such file or directory.
> ```
> 
> I think this is something Luca just fixed for other tests. Do we need to add something to the inputs for these tests?
> 
> The SideSetLaplacian_3D test is failing with the following error:
> 
> ```
> 99: ************************************************************************
> 99: -- Nonlinear Solver Step 49 --
> 99: ||F|| = 2.500e-01  step = 1.000e+00  dx = 0.000e+00
> 99: ************************************************************************
> 99:
> 99: Ifpack2::ILUT::Compute: zero pivot encountered! Replacing with rownorm and continuing...(You may need to set the parameter 'fact: absolute threshold'.)
> 99: Ifpack2::ILUT::Compute: zero pivot encountered! Replacing with rownorm and continuing...(You may need to set the parameter 'fact: absolute threshold'.)
> 99: Ifpack2::ILUT::Compute: zero pivot encountered! Replacing with rownorm and continuing...(You may need to set the parameter 'fact: absolute threshold'.)
> 99: Ifpack2::ILUT::Compute: zero pivot encountered! Replacing with rownorm and continuing...(You may need to set the parameter 'fact: absolute threshold'.)
> 99:
> 99:   *******************************************************
> 99:   ***** Belos Iterative Solver:  Block Gmres
> 99:   ***** Maximum Iterations: 100
> 99:   ***** Block Size: 1
> 99:   ***** Residual Test:
> 99:   *****   Test 1 : Belos::StatusTestImpResNorm<>: (2-Norm Res Vec) / (2-Norm Res0), tol = 0.0001
> 99:   *******************************************************
> 99:   Iter   0, [ 1] :    1.000000e+00
> 99:   Belos::BlockGmresSolMgr::solve(): Warning! NaN has been detected!
> 99: WARNING: NOX::Direction::Newton::compute() - Linear solve failed to achieve convergence - using the step anyway since "Rescue Bad Newton Solve" is true
> 99: NOX::Solver::LineSearchBased::iterate - using recovery step for line search
> ```
> 
> Not sure why this would fail but not the 2D version.

I'll fix the tests failing for the error with loading the mesh.
As for the SideSetLaplacian_3D test, it must be some uninitialized variable or something like that. On my workstation it converges in 1 iteration. I'll see if I can find the culprit, but I might go ahead and push the PR anyway, and address the error later.

File: tests/landIce/FO_GIS/CMakeLists.txt (hunk 1 above)
Comment (mperego) [reply tests/landIce/FO_GIS/CMakeLists.txt:92]: Yes, I'll remove one.

**Comment by mperego** [comment]
OK, I have renamed the tests to remove references to Tpetra and hopefully fixed the exudus issues in the test files.
I could not reproduce the issue with  `SideSetLaplacian_3D` on weaver. But I'd rather push the PR and we look at that later.

**Review by bartgol — APPROVED** [review]
Thanks for the great cleanup work Mauro! I love PRs with more red than green... :)

**Review by mcarlson801 — APPROVED** [review]
Looks good to me!

**Comment by mperego** [comment]
Thanks @bartgol and @mcarlson801 for your feedback!

**Pull request merged**